packages/flutter/test/rendering/object_test.dart

```diff
@@ -255,6 +334,39 @@ class TestRenderObject extends RenderObject {
   }
 }
 
+class LeaderLayerRenderObject extends RenderObject {
+  LeaderLayerRenderObject();
+
+  LayerLink? layerLink;
+
+  @override
+  bool isRepaintBoundary = true;
```

Comment:
Just to double check: There isn't a problem if we are not a repaint boundary?

Reply:
It will still be a problem.

The reason i made it a repaint boundary is for easy testing. If it is not a repaint boundary, the markNeedsPaint will go through the ancestor chain to find the nearest repaint boundary and markNeedsPaint on it. That means I would have to create a parent repaint boundary in the test anyway.

Reply:
But that case is also fixed by this fix, right?

Reply:
Yes, In the layer tree, the non-repaint boundary renderobject does not contributing additional node, so there is no difference between layer added by non-repaint boundary renderobject and layer added by the nearest repaint boundary.